### PR TITLE
feat(VETS-CPC-904): change workdays of a vet to an enum

### PR DESCRIPTION
**Pull request by KarinaSofia**
JIRA: [link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/CPC-904)
## Context:
Modified the workday field and modified the front end to be checkboxes when selecting the workdays
## Changes
- modified all vetdtos to switch workdays from a string to an enum
- created the workday enum
- modified the details page to show a list of the workdays of a vet
- modified the form page to have checkboxes instead of an input box when modifying or adding a vet

### Before 
image
image
image

### After
image
image
image

### CODE REVIEW

File: api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java [hunk 1]
```diff
@@ -425,7 +425,7 @@ public Flux<VetDTO> getAllVets() {
         return vetsServiceClient.getVets();
     }
 
-    @IsUserSpecific(idToMatch = {"vetId"})
+    @SecuredEndpoint(allowedRoles = {Roles.ANONYMOUS})
```
Comment (Iantomasi) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:428]: Hey Karina, I know that Dylan is the one who implemented the secure endpoints but just to double check, what is ANONYMOUS role?

File: visits-service-new/src/test/java/com/petclinic/visits/visitsservicenew/PresentationLayer/VisitControllerUnitTest.java [hunk 2]
```diff
@@ -50,7 +50,7 @@ class VisitControllerUnitTest {
 
 
     Set<SpecialtyDTO> set= new HashSet<>();
-
+    Set<Workday> workdaySet = new HashSet<>();
```
Comment (Iantomasi) [inline visits-service-new/src/test/java/com/petclinic/visits/visitsservicenew/PresentationLayer/VisitControllerUnitTest.java:53]: Smart value change Karina. Modifications made to visit-service-new look good as well

**Review by Iantomasi — APPROVED** [review]

File: api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java (hunk 1 above)
Comment (KarinaSofia) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:428]: ANONYMOUS is anyone!

**Review by Sprytte — APPROVED** [review]
Everything looks good! The switch to Enum was properly implemented and its usage was appropriately modified,  including in other services and tests.

**Pull request merged**